pygments/lexers/usd.py

```diff
@@ -0,0 +1,136 @@
+                    token.Name.Keyword.Tokens,
+                    token.Whitespace,
+                    token.Operator,
+                ),
+            ),
+            (
+                r"(custom){_WHITESPACE}{}(\s+){}(\s*)(=)".format(
+                    _TYPE, _BASE_ATTRIBUTE, _WHITESPACE=_WHITESPACE
+                ),
+                lexer.bygroups(
+                    token.Keyword.Token,
+                    token.Whitespace,
+                    token.Keyword.Type,
+                    token.Whitespace,
+                    token.Name.Attribute,
+                    token.Generic,
+                    token.Name.Keyword.Tokens,
+                    token.Whitespace,
+                    token.Operator,
+                ),
+            ),
+            (
+                r"(uniform){_WHITESPACE}{}(\s+){}(\s*)(=)".format(
+                    _TYPE, _BASE_ATTRIBUTE, _WHITESPACE=_WHITESPACE
+                ),
+                lexer.bygroups(
+                    token.Keyword.Token,
+                    token.Whitespace,
+                    token.Keyword.Type,
+                    token.Whitespace,
+                    token.Name.Attribute,
+                    token.Generic,
+                    token.Name.Keyword.Tokens,
+                    token.Whitespace,
+                    token.Operator,
+                ),
+            ),
+            (
+                r"{}{_WHITESPACE}{}(\s*)(=)".format(
+                    _TYPE, _BASE_ATTRIBUTE, _WHITESPACE=_WHITESPACE
+                ),
+                lexer.bygroups(
+                    token.Keyword.Type,
+                    token.Whitespace,
+                    token.Name.Attribute,
+                    token.Generic,
+                    token.Name.Keyword.Tokens,
+                    token.Whitespace,
+                    token.Operator,
+                ),
+            ),
+        ]
+        + _keywords(_usd_builtins.KEYWORDS, token.Keyword.Tokens)
+        + _keywords(_usd_builtins.SPECIAL_NAMES, token.Name.Builtins)
+        + _keywords(_usd_builtins.COMMON_ATTRIBUTES, token.Name.Attribute)
+        + [(r"\b\w+:[\w:]+\b", token.Name.Attribute)]
+        + _keywords(_usd_builtins.OPERATORS, token.Operator)  # more attributes
+        + [(type_ + r"\[\]", token.Keyword.Type) for type_ in _usd_builtins.TYPES]
+        + _keywords(_usd_builtins.TYPES, token.Keyword.Type)
+        + _PUNCTUATION
+        + [
+            ("#.*?$", token.Comment.Single),
+            (",", token.Generic),
+            (";", token.Generic),  # ";"s are allowed to combine separate metadata lines
+            ("=", token.Operator),
+            ("[-]?([0-9]*[.])?[0-9]+", token.Number),
+            (r"'''(?:.|\n)*?'''", token.String),
+            (r'"""(?:.|\n)*?"""', token.String),
+            (r"'.*'", token.String),
+            (r'".*"', token.String),
+            (r"<(\.\./)*([\w/]+|[\w/]+\.\w+[\w:]*)>", token.Name.Namespace),
+            (r"@.*@", token.String.Interpol),
```

Comment:
these rules usually have to be written with `.*?` to avoid greedy highlighting of a line like `@firststring@ something else @secondstring@`.

Same for double- and single quoted strings.

Reply:
Oh, didn't know about that. Thanks - done on all accounts + added a test for this